src/Microsoft.AspNet.Mvc.Core/ViewComponents/IViewComponentDescriptorCollectionProvider.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNet.Mvc.ViewComponents
+{
+    /// <summary>
+    /// Provides the currently cached collection of <see cref="ViewComponentDescriptor"/>.
+    /// </summary>
+    /// <remarks>
+    /// The default implementation, does not update the cache, it is up to the user
```

Comment:
nit: punctuation makes this hard to read.  think you mean "The default implementation does not update the cache; it ..." or "The default implementation does not update the cache. It".  similar comment for next paragraph.